File: BlackJackInPy/Deck.py
class Deck:

    def __init__(self, deck_type):
        self.my_deck = []
        if deck_type == "Bank":
            for index in range(9):
                self.my_deck.append([index + 2, 4])
            # "Jack"
            self.my_deck.append([10, 4])
            # "Queen"
            self.my_deck.append([10, 4])
            # "King"
            self.my_deck.append([10, 4])
            # "Ace"
            self.my_deck.append([0, 4])
        elif deck_type == "Dealer" or deck_type == "Player":
            for index in range(9):
                self.my_deck.append([index + 2, 0])
            # "Jack"
            self.my_deck.append([10, 0])
            # "Queen"
            self.my_deck.append([10, 0])
            # "King"
            self.my_deck.append([10, 0])
            # "Ace"
            self.my_deck.append([0, 0])
        else:
            raise RuntimeError("Just Bank/Player/Dealer!")

    def get_points(self):
        points = 0
        for index in range(12):
            points += self.my_deck[index][0] * self.my_deck[index][1]
        # Ace
        if (points+self.my_deck[12][1]*11) <= 21:
            points += 11 * self.my_deck[12][1]
        else:
            points += 1 * self.my_deck[12][1]
        return points

    def get_new_card(self, position):
        self.my_deck[position][1] += 1

    def remove_card(self, position):
        if self.my_deck[position][1] > 0:
            self.my_deck[position][1] -= 1
        else:
            raise RuntimeError("Out from that card")

    def my_cards(self):
        for index in range(9):
            if self.my_deck[index][1] > 0:
                print(str(self.my_deck[index][0]) + " X " + str(self.my_deck[index][1]))
        if self.my_deck[9][1] > 0:
            print("Jack X " + str(self.my_deck[9][1]))
        if self.my_deck[10][1] > 0:
            print("Queen X " + str(self.my_deck[10][1]))
        if self.my_deck[11][1] > 0:
            print("King X " + str(self.my_deck[11][1]))
        if self.my_deck[12][1] > 0:
            print("Ace X " + str(self.my_deck[12][1]))





'''
#ez nem mukodik
class PlayerDeck(Deck):

    def __init__(self):
        Deck.__init__(self)
        for index in range(9):
            self.my_deck.append([index + 2, 0])
        # "Jack"
        self.my_deck.append([10, 0])
        # "Queen"
        self.my_deck.append([10, 0])
        # "King"
        self.my_deck.append([10, 0])
        # "Ace"
        self.my_deck.append([0, 0])


class BankDeck(Deck):

    def __init__(self):
        Deck.__init__(self)
        for index in range(9):
            self.my_deck.append([index + 2, 4])
        # "Jack"
        self.my_deck.append([10, 4])
        # "Queen"
        self.my_deck.append([10, 4])
        # "King"
        self.my_deck.append([10, 4])
        # "Ace"
        self.my_deck.append([0, 4])

'''

File: BlackJackInCpp/BlackJackInCpp/Deck.cpp
#include "stdafx.h"
#include "Deck.h"


Deck::Deck(string dackType)		//kiszervezheto kulon osztalyba, a haszon elhanyagolhato
{
	if (dackType == "Bank")
	{
		for (int i = 0; i<9; ++i) {		//2-10
			myDeck.push_back(make_pair(4, i + 2));
			//mennyiseg es ertek			
		}
		//Jack
		myDeck.push_back(make_pair(4, 10));

		//Queen
		myDeck.push_back(make_pair(4, 10));

		//King
		myDeck.push_back(make_pair(4, 10));;

		//Ace
		myDeck.push_back(make_pair(4, 0));

	}
	else if (dackType == "Player" || dackType == "Dealer")
	{
		for (int i = 0; i<9; ++i) {		//2-10
			myDeck.push_back(make_pair(0, i + 2));
			//mennyiseg es ertek			
		}
		//Jack
		myDeck.push_back(make_pair(0, 10));

		//Queen
		myDeck.push_back(make_pair(0, 10));

		//King
		myDeck.push_back(make_pair(0, 10));;

		//Ace
		myDeck.push_back(make_pair(0, 0));
	}
	else
		throw std::runtime_error("Only Player ,Dealer or Bank is possible");
}

int Deck::getPoints() {
	int points = 0;
	vector<pair<int, int>>::iterator iter_out;
	int i = 0;
	for (iter_out = myDeck.begin(); i<12; ++iter_out, ++i) {
		points += iter_out->first*iter_out->second;
	}
	//Aszra kulon szabaly van
	if (points + 11 * myDeck.at(12).first <= 21)
		points += 11 * myDeck.at(12).first;
	else
		points += 1 * myDeck.at(12).first;

	return points;
}

void Deck::getNewCard(int num) {
	myDeck.at(num).first++;
}

void Deck::removeCard(int num) {
	if (myDeck.at(num).first != 0)
		myDeck.at(num).first--;
	else
		throw runtime_error("nincs tobb ilyen lap a pakliban");
}

void Deck::myCards() {
	vector<pair<int, int>>::iterator iter_out;
	int i = 0;
	for (iter_out = myDeck.begin(); i<9; ++iter_out, ++i) {
		if (iter_out->first != 0)
			cout << i + 2 << " X " << iter_out->first << endl;
	}
	if (myDeck.at(9).first != 0)
		cout << "Jack" << " X " << myDeck.at(9).first << endl;
	if (myDeck.at(10).first != 0)
		cout << "Queen" << " X " << myDeck.at(10).first << endl;
	if (myDeck.at(11).first != 0)
		cout << "King" << " X " << myDeck.at(11).first << endl;
	if (myDeck.at(12).first != 0)
		cout << "Ace" << " X " << myDeck.at(12).first << endl;
}

Deck::~Deck()
{
}
